### DISCUSSION TITLE
INL partners with Coreform to improve open-source modeling and simulation tool MOOSE

### DISCUSSION THREAD

**Discussion by aeslaughter** [idaholab/moose#16774] [News]
https://inl.gov/article/inl-partners-with-coreform-to-improve-open-source-modeling-and-simulation-tool-moose/

**Comment by mangerij** [comment]
Very cool, does this mean there will be some sort of MOOSE system in the works that uses Coreform's tool to optimize simulations?

**Reply by permcody** [reply]
Not in the short term. The immediate focus will be making Cubit a good pre/post processing tool for use with MOOSE.

**Comment by WilkAndy** [comment]
Details?  This sounds important, but i'm not sure exactly what it means to me.

a

________________________________________

**Reply by permcody** [reply]
Cubit now has a limited free product available. We are continuing to negotiate with Coreform to negotiate both of our business models.

**Reply by GregVernon** [reply]
@WilkAndy To add to what Cody has mentioned, we at Coreform have been collaborating with INL to get smooth-splines supported by MOOSE - via Bezier extraction.  We've been helping/partnering with Roy Stogner to support these elements in libMesh ([see GitHub issue](https://github.com/libMesh/libmesh/issues/2179)), developing a solution for Exodus to support Bezier extraction data-structures ([see GitHub issue](https://github.com/gsjaardema/seacas/issues/189)), we've added support for Bezier cells (the piecewise components of splines) in VTK and they're now supported in ParaView 5.8 and later ([see KitWare blog post](https://blog.kitware.com/implementation-of-rational-bezier-cells-into-vtk/)), and we will soon support exporting Bezier extraction from Coreform Cubit - as a native capability. 

What does this mean?  Bezier extraction is _the_ method that links traditional FEM with spline-based FEM.  Thus support of Bezier extraction throughout the FEM workflow (Meshing tool: Cubit; Mesh file: Exodus; FEM library: libMesh; Post-processing: ParaView) enables spline-based FEM to a broad audience. You can think of spline-based FEM as a generalization of FEM that allows for use of smooth basis functions.  Smoothness of the basis can provide benefits like having defined derivatives (face-normals) at element boundaries - making things like contact, ray-tracing, etc. more well-behaved.  Smooth splines have been recognized as being useful for FEM at least since [this insightful review paper by Gilbert Strang (1973) of Richard Courant's development of FEM](https://pdfs.semanticscholar.org/6660/d81db921ebe714b16878616ccca957ef89d5.pdf).  We're hopeful that our collaboration will finally enable practical FEM analyses using spline basis functions - and implemented within MOOSE!

**Reply by fdkong** [reply]
It is an exciting path
